ain/utils.py

```diff
@@ -322,13 +327,67 @@ def mnemonicToPrivatekey(mnemonic: str, index: int = 0) -> bytes:
     path = AIN_HD_DERIVATION_PATH + f"{index}"
     return bip32.get_privkey_from_path(path)
 
-# TODO(kriii): implement this function.
-def encryptWithPublicKey():
-    pass
+def encryptWithPublicKey(publicKey: Union[bytes, str], message: str) -> ECDHEncrypted:
+    """
+    Encrypts message with publicKey.
+    """
+    if type(publicKey) is str:
+        publicKeyBytes = bytes.fromhex(publicKey)
+    elif type(publicKey) is bytes:
+        publicKeyBytes = publicKey
+    else:
+        raise ValueError("Invalid public key type")
 
-# TODO(kriii): implement this function.
-def decryptWithPrivateKey():
-    pass
+    if len(publicKeyBytes) == 64:
+        publicKeyBytes = bytes([4]) + publicKeyBytes
+
+    pbK = PublicKey(publicKeyBytes)
```

Comment:
question) Is there a link/docs you consulted while implementing these functions? If so, it would be nice if you left them as a note~

Reply:
e.g. `NOTE(kriii): Referenced https://github.com/bitchan/eccrypto/blob/ad39c5d117edcd397420696954279e428eae1bb2/index.js
https://cryptobook.nakov.com/asymmetric-key-ciphers/ecies-public-key-encryption
https://github.com/ecies/py/blob/b505e0bd0bcc06282ec45a9ddfe9268a6021426f/ecies/utils.py` ?

Reply:
I added only about eccrypto, because there is nothing left on code about corresponds with the following two refs.